### PR TITLE
316 ensure worker signature is in all scripts that case/note

### PR DESCRIPTION
**Pull request by MiKaylaH22**
-For each script: reviewed changelog, updated field for worker_signature in initial dialog, made worker_signature mandatory, and ensured case note was written correctly.
In specialty scripts, Shelter and DEU the team name was listed in lieu of the worker signature and the worker signature has been added and the the team name retained.

### CODE REVIEW

**Comment by IlseFerris** [comment]
I'll be reviewing this on Monday.

File: dail/fmed-deduction.vbs [hunk 1]
```diff
@@ -69,7 +70,8 @@ EndDialog
 
 Do
```
Comment (IlseFerris) [inline dail/fmed-deduction.vbs:71]: The check_for_password needs to be in it's own do...loop. Add a do...loop for worker_signature

File: dail/citizenship-verified.vbs [hunk 2]
```diff
@@ -147,16 +146,20 @@ Next
...
+    CancelButton 135, 25, 50, 15
 EndDialog
 
-Dialog Dialog1 
-Cancel_without_confirmation
+	Do
+		err_msg = ""
+		Dialog Dialog1
+		cancel_without_confirmation
+		IF worker_signature = "" THEN err_msg = err_msg & vbCr & "* Please sign your case note."
```
Comment (IlseFerris) [inline dail/citizenship-verified.vbs:161]: Add password function in own do...loop

File: admin/individual-appointment-letter.vbs [hunk 3]
```diff
@@ -62,30 +63,28 @@ Call check_for_MAXIS(TRUE)
...
     err_msg = ""
-
     Dialog Dialog1
     cancel_without_confirmation
-
-    If len(MAXIS_case_number) >8 Then err_msg = err_msg & vbNewLine & "* Case numbers should not be more than 8 numbers long."
-    If IsNumeric(MAXIS_case_number) = FALSE Then err_msg = err_msg & vbNewLine & "* Check the case number, it appears to be invalid."
-
+	Call validate_MAXIS_case_number(err_msg, "*")
+	IF worker_signature = "" THEN err_msg = err_msg & vbCr & "* Please sign your case note."
```
Comment (IlseFerris) [inline admin/individual-appointment-letter.vbs:81]: Add password handling in it's own do....loop

File: admin/individual-nomi.vbs [hunk 4]
```diff
@@ -68,23 +69,22 @@ EMConnect ""
...
-
     Dialog Dialog1
-    If buttonpressed = Cancel Then script_end_procedure_with_error_report("")
-
-    If len(MAXIS_case_number) >8 Then err_msg = err_msg & vbNewLine & "* Case numbers should not be more than 8 numbers long."
-    If IsNumeric(MAXIS_case_number) = FALSE Then err_msg = err_msg & vbNewLine & "* Check the case number, it appears to be invalid."
-
+  	cancel_without_confirmation
+	Call validate_MAXIS_case_number(err_msg, "*")
+ 	IF worker_signature = "" THEN err_msg = err_msg & vbCr & "* Please sign your case note."
```
Comment (IlseFerris) [inline admin/individual-nomi.vbs:87]: Add password handling in it's own do...loop

File: dail/new-hire-ndnh.vbs [hunk 5]
```diff
@@ -240,6 +241,7 @@ IF match_answer_droplist = "NO-RUN NEW HIRE" THEN
 	    	Dialog dialog1
 	    	cancel_confirmation
 	    	MAXIS_dialog_navigation
+			IF worker_signature = "" THEN err_msg = err_msg & vbCr & "* Please sign your case note."
```
Comment (IlseFerris) [inline dail/new-hire-ndnh.vbs:244]: Err_msg isn't set in the do...loop. No msgbox code to give the worker_signature error message.

File: dail/new-hire.vbs [hunk 6]
```diff
@@ -199,6 +200,7 @@ DO
     	Dialog dialog1
     	cancel_confirmation
     	MAXIS_dialog_navigation
+		IF worker_signature = "" THEN err_msg = err_msg & vbCr & "* Please sign your case note."
```
Comment (IlseFerris) [inline dail/new-hire.vbs:203]: Same as NEW HIRE NDNH: Err_msg isn't set in the do...loop. No msgbox code to give the worker_signature error message.

File: dail/postponed-expedited-snap-verifications.vbs [hunk 7]
```diff
@@ -97,10 +98,11 @@ BeginDialog Dialog1, 0, 0, 401, 70, "Verifications Needed"
 EndDialog
 
 'Runs the dialog to allow workers to sign and to list verifications needed
-Do 
+Do
     Do
     	Dialog Dialog1
-    	Cancel_without_confirmation
+    	cancel_without_confirmation
+		IF worker_signature = "" THEN err_msg = err_msg & vbCr & "* Please sign your case note."
```
Comment (IlseFerris) [inline dail/postponed-expedited-snap-verifications.vbs:105]: Same as the NEW HIRE's: Err_msg isn't set in the do...loop. No msgbox code to give the worker_signature error message.

File: deu/overpayment-claim-entered.vbs [hunk 8]
```diff
@@ -185,7 +188,8 @@ Do
 		IF HC_to = "" THEN err_msg = err_msg & vbNewLine &  "* Please enter the month and year overpayment ended."
 		IF HC_claim_amount = "" THEN err_msg = err_msg & vbNewLine &  "* Please enter the amount of claim."
 	END IF
-    	IF EVF_used = "" then err_msg = err_msg & vbNewLine & "* Please enter verification used for the income received. If no verification was received enter N/A."
+    IF EVF_used = "" then err_msg = err_msg & vbNewLine & "* Please enter verification used for the income received. If no verification was received enter N/A."
+	IF worker_signature = "" THEN err_msg = err_msg & vbCr & "* Please sign your case note."
 	IF err_msg <> "" THEN MsgBox "*** NOTICE!!! ***" & vbNewLine & err_msg & vbNewLine		'error message including instruction on what needs to be fixed from each mandatory field if incorrect
 LOOP UNTIL err_msg = ""
```
Comment (IlseFerris) [inline deu/overpayment-claim-entered.vbs:194]: Add password handling and do....loop

File: notes/citizenship-identity-verified.vbs [hunk 9]
```diff
@@ -119,6 +120,7 @@ Do
 		can_move_on = True
 	End if
 	If can_move_on = False then MsgBox "You must select a cit and ID proof for each client whose name you've typed."
+	IF worker_signature = "" THEN err_msg = err_msg & vbCr & "* Please sign your case note."
 Loop until can_move_on = True
```
Comment (IlseFerris) [inline notes/citizenship-identity-verified.vbs:124]: Add password handling in it's own do...loop

File: notes/drug-felon.vbs [hunk 10]
```diff
@@ -236,6 +239,7 @@ Case "Initial Information Not Received"
 		cancel_confirmation
 		If clt_to_update = "Select One..." Then err_msg = err_msg & vbNewLine & "Please pick the client you are processing DFLN information for."
 		If err_msg <> "" Then MsgBox "Please resolve to continue:" & vbNewLine & err_msg
+	IF worker_signature = "" THEN err_msg = err_msg & vbCr & "* Please sign your case note."
```
Comment (IlseFerris) [inline notes/drug-felon.vbs:242]: Add password handling function in it's own do...loop

File: notes/grh-non-hrf-postpay.vbs [hunk 11]
```diff
@@ -518,6 +520,7 @@ DO
 			IF err_msg <> "" AND ButtonPressed = -1 THEN MsgBox "*** NOTICE!!! ***" & vbCr & err_msg & vbCr & vbCr & "Please resolve for the script to continue."
 		LOOP UNTIL err_msg = "" OR ButtonPressed = previous_button
 	LOOP WHILE ButtonPressed = previous_button
+	IF worker_signature = "" THEN err_msg = err_msg & vbCr & "* Please sign your case note."
```
Comment (IlseFerris) [inline notes/grh-non-hrf-postpay.vbs:523]: worker signature is outside of the do...loop with the err_msg. Update to within the err_msg do....loop.

File: notes/ltc-cola-summary.vbs [hunk 12]
```diff
@@ -301,7 +305,7 @@ If approval_summary_check = 1 then
     EndDialog
 
     Do
-        Dialog Dialog1 
+        Dialog Dialog1
```
Comment (IlseFerris) [inline notes/ltc-cola-summary.vbs:308]: worker_signature is in dialog, but not mandatory in the do....loop. Either remove variable from dialog or add handling and error messaging.

File: notes/ltc-cola-summary.vbs [hunk 13]
```diff
@@ -518,13 +522,13 @@ If income_summary_check = 1 then
       Text 5, 160, 45, 10, "Other notes:"
       Text 135, 175, 60, 15, "Worker signature:"
     EndDialog
-    
-    Do 
+
```
Comment (IlseFerris) [inline notes/ltc-cola-summary.vbs:525]: Same as other dialog: worker_signature is in dialog, but not mandatory in the do....loop. Either remove variable from dialog or add handling and error messaging.

File: deu/match-cleared.vbs [hunk 14]
```diff
@@ -911,6 +911,7 @@ IF resolution_status = "CC-Overpayment Only" or HC_OP_checkbox = CHECKED THEN
 END IF
 CALL write_bullet_and_variable_in_case_note("Other Notes", other_notes)
 CALL write_variable_in_case_note("----- ----- ----- ----- ----- ----- -----")
+CALL write_variable_in_case_note(worker_signature)
```
Comment (IlseFerris) [inline deu/match-cleared.vbs:914]: Do....loop on lines 85-117 does not have password handling. While this line was only added to the CASE/NOTE on this script as part of the issue, the mandatory handling associated with this issue is affected by not having a complete do....loop series. Please update.

**Review by IlseFerris — CHANGES_REQUESTED** [review]
Globally I would use trim(worker_signature) in the mandatory fields, BUT unless you need to go back and make updates on any of the files DO NOT make these updates now. 

All updates have a specific comment in the code. Over all looks good, but there is some  do...loops handling/updates that are needed.

File: notes/grh-non-hrf-postpay.vbs (hunk 11 above)
Comment (MiKaylaH22) [reply notes/grh-non-hrf-postpay.vbs:523]: was going to add 
    MAXIS_footer_month = right("00" & MAXIS_footer_month, 2)
    MAXIS_footer_year = right("00" & MAXIS_footer_year, 2)
    MAXIS_footer_month_confirmation	'function will check the MAXIS panel footer month/year vs. the footer month/year in the dialog, and will navigate to the dialog month/year if they do not match.

but I didnt want it to be out of scope.. and then I had a question so I left it alone

File: notes/drug-felon.vbs [hunk 15]
```diff
@@ -44,7 +44,8 @@ changelog = array()
 
 'INSERT ACTUAL CHANGES HERE, WITH PARAMETERS DATE, DESCRIPTION, AND SCRIPTWRITER. **ENSURE THE MOST RECENT CHANGE GOES ON TOP!!**
 'Example: call changelog_update("01/01/2000", "The script has been updated to fix a typo on the initial dialog.", "Jane Public, Oak County")
-call changelog_update("12/29/2016", "Adding functionality to case note the return of required documenttion when a drug felon match is reported.", "Casey Love, Ramsey County")
+
+CALL changelog_update("09/19/2022", "Update to ensure Worker Signature is in all scripts that CASE/NOTE.", "MiKayla Handley, Hennepin County") '#316call changelog_update("12/29/2016", "Adding functionality to case note the return of required documenttion when a drug felon match is reported.", "Casey Love, Ramsey County")
```
Comment (IlseFerris) [inline notes/drug-felon.vbs:48]: 12/26/2016 changelog is commented out. Restore this back into the changelog.

**Review by IlseFerris — CHANGES_REQUESTED** [review]
Just one quick change on a changelog, then I'll approve it.

**Review by IlseFerris — APPROVED** [review]
I approve thee!

**Pull request merged**